### ISSUE TITLE
Build error using release source archive 

### ISSUE THREAD

**Issue by gobbledygoober** [mwpenny/portal64-still-alive#85]
#### Version of the game
 - Version v0.16.0
 - 9e1dfa3

#### Original Hardware or Emulator?
N/A - Build issue

#### What is the bug?
For the v0.16.0 release archives, build attempts result in a failure:
```
mips-n64-gcc -c -mabi=32 -ffreestanding -mfix4300 -G 0     -DDEBUG -DGAME_VERSION=\"\" -g -Werror -Wall -DF3DEX_GBI_2 -DSCENE_SCALE=128 -I/usr/include/n64/PR -Isrc/ -I/opt/crashsdk/mips64-elf/include   -I/usr/include/n64 -Os    -MM src/menu/landing_menu.c build/assets/materials/ui.h build/src/audio/clips.h build/version.txt -MF "build/src/menu/landing_menu.d" -MT"build/src/menu/landing_menu.o"
mips-n64-gcc: warning: build/version.txt: linker input file unused because linking not done
mips-n64-gcc: error: build/version.txt: linker input file not found: No such file or directory
make[1]: *** [Makefile:155: build/src/menu/landing_menu.o] Error 1
make[1]: Leaving directory '/mnt/c/Users/username/portal64/portal64-still-alive-0.16.0'
make: *** [Makefile:92: english_audio] Error 2
```

The required `build/version.txt` is missing when built from release archives. The `GAME_VERSION` variable expands to an empty string in this case because `git` commands are used in the assignment, and release archives are not git repositories.

As a workaround, the code from the equivalent 9e1dfa3 commit can be checked out, or https://github.com/mwpenny/portal64-still-alive/blob/master/Makefile#L23 can be modified with an additional conditional bash assignment. 

For example, the original line:
```
# Use tag name if a tag exists, otherwise use commit hash
GAME_VERSION            :=  $(shell git describe --tags --exact-match HEAD 2>/dev/null || git rev-parse --short HEAD)
```
And modification for the v0.16.0 release:
```
# Use tag name if a tag exists, otherwise use commit hash
GAME_VERSION            :=  $(shell git describe --tags --exact-match HEAD 2>/dev/null || git rev-parse --short HEAD 2>/dev/null || echo 0.16.0)
```
Ideas for long-term solutions:
* Pre-populate build/version.txt for release archives and check if it's already present before creating it in the `Makefile`
* Release preparation can use `sed` to edit the `Makefile` with the version
* The `README.md` can be modified with build instructions to set the variable for releases, e.g. `GAME_VERSION=0.16.0 make`, with corresponding `Makefile` changes 

On a related note, I recently encountered a similar issue building Wireshark. Their solution is a little unintuitive and I wish it were a bit more documented. An archive of the source must be created (using tar.xz vs the [zip and tar.gz formats distributed on github](https://github.com/wireshark/wireshark/tags)), and copied to the build directory with a specific filename syntax. If the directory is found to not be a git repository, the version is read from the filename of the new archive:
https://gitlab.com/wireshark/wireshark/-/blob/master/packaging/source/git-export-release.sh.in#L65-70

I do admire how Wireshark controls a script that can be used to facilitate the creation of a release source archive from a git repository. If the first bulleted solution listed above is used in portal64-stil-alive, such a script could create the `build/version.txt`.

#### Steps to reproduce the behavior:
1. Download the tar.gz archive from https://github.com/mwpenny/portal64-still-alive/releases/tag/v0.16.0
2. Extract
3. Follow README.md build setup
4. Run `make`

#### Screenshots
N/A

#### Additional context
Building under wsl2 with Ubuntu 22.04.3 LTS

**Comment by mwpenny** [comment]
Thanks for the details. I agree supporting building from repo snapshots is reasonable. I'll look into this.

**Comment by gobbledygoober** [comment]
Finally had a chance to try this with a zip download from today. Worked great. Afterwards, I checked `version.txt` as a sanity check:
```
$ cat build/version.txt
ad38717b
```
That is a pretty slick implementation! I wonder if GitLab and GitTea's "download archive" mechanism also uses git-archive and supports this as well. 

I appreciate you working on this so quickly and for keeping the issue open until I was able to provide feedback.
